src/axom/inlet/Inlet.cpp

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ *******************************************************************************
+ * \file Inlet.cpp
+ *
+ * \brief This file contains the class implementation of Inlet, the main class
+ *        for the Inlet component.
+ *******************************************************************************
+ */
+
+#include "axom/inlet/Inlet.hpp"
+
+#include "fmt/fmt.hpp"
+#include "axom/slic.hpp"
+
+namespace axom
+{
+namespace inlet
+{
+
+std::shared_ptr<Table> Inlet::addTable(const std::string& name,
+                                       const std::string& description)
+{
+  return std::make_shared<Table>(name, description, m_reader, m_sidreRootGroup);
+}
+
+std::shared_ptr<Field> Inlet::addBool(const std::string& name,
+                                      const std::string& description)
+{
+  return m_globalTable->addBool(name, description);
+}
+
+std::shared_ptr<Field> Inlet::addDouble(const std::string& name,
+                                        const std::string& description)
+{
+  return m_globalTable->addDouble(name, description);
+}
+
+std::shared_ptr<Field> Inlet::addInt(const std::string& name,
+                                     const std::string& description)
+{
+  return m_globalTable->addInt(name, description);
+}
+
+std::shared_ptr<Field> Inlet::addString(const std::string& name,
+                                        const std::string& description)
+{
+  return m_globalTable->addString(name, description);
+}
+
+
+//-------------------------------------------------
+//   Get values out of the datastore
+//-------------------------------------------------
+
+axom::sidre::View* Inlet::baseGet(const std::string& name)
+{
+  SLIC_ASSERT_MSG(m_sidreRootGroup != nullptr, "Inlet's Sidre Datastore Group not set");
+
+  // All data hangs under the group's name
+  if (!m_sidreRootGroup->hasGroup(name))
+  {
+    return nullptr;
```

Comment:
It might be nice to provide the user with a log of what went wrong.

Perhaps with a way to modify the verbosity when desired?

Reply:
True, but what if the value is optional?  Should I add a hasValue maybe to allow them to check before hand?